python/ext-libs/owslib/csw.py

```diff
@@ -267,7 +267,15 @@ def getrecordbyid(self, id=[], esn='full', outputschema=namespaces['csw'], forma
             'id': ','.join(id),
         }
 
-        self.request = '%s%s' % (bind_url(self.url), urlencode(data))
+        l = []
+        data = data.items()
+        for k, v in data:
+            k = str(k)
```

Comment:
`str` should not be used for strings.

 * Remove it if you are sure that there is a string returned always anyway
 * Change it to `unicode()` if you need to convert it to a string